File: Leetcode/477. Total Hamming Distance/total_hamming_distance.py
class Solution:
    def totalHammingDistance(self, nums) -> int:
        res = 0
        for i in range(32):
            count = 0
            for num in nums:
                count += (num>>i)&1
            # ones * zeros
            res += count * (len(nums)-count)
        return res
    
sol = Solution()
print(sol.totalHammingDistance([4,14,2]))

File: Leetcode/477. Total Hamming Distance/total_hamming_distance.rs
use std::collections::{HashSet, HashMap};

struct Solution{}

impl Solution {
    pub fn total_hamming_distance(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        for i in 0..32{
            let mut count = 0;
            for num in &nums{
                count += (num>>i)&1;
            }
            res += count * (nums.len() as i32 -count)
        }
        res
    }
}
fn main() {
    let res = Solution::total_hamming_distance(vec![4,14,2]);
    println!("{:?}",res);
}
